File: address-insight-us-nodejs-examples/get_address_insight_rest_sdk_example.js
import { GetAddressInsightClient } from '../address-insight-us-nodejs/REST/get_address_insight_rest.js';

async function GetAddressInsightClientGo(licenseKey, isLive) {
    console.log("\n---------------------------------------------------");
    console.log("Address Insight - US - GetAddressInsight - REST SDK");
    console.log("---------------------------------------------------");

    const businessName = "";
    const address1 = "26 S Chestnut";
    const address2 = "";
    const city = "Ventura";
    const state = "CA";
    const zip = "93033";
    const testType = "";
    const timeoutSeconds = 15;

    console.log("\n* Input *\n");
    console.log(`Business Name  : ${businessName}`);
    console.log(`Address1       : ${address1}`);
    console.log(`Address2       : ${address2}`);
    console.log(`City           : ${city}`);
    console.log(`State          : ${state}`);
    console.log(`Zip            : ${zip}`);
    console.log(`Test Type      : ${testType}`);
    console.log(`License Key    : ${licenseKey}`);
    console.log(`Is Live        : ${isLive}`);
    console.log(`Timeout Seconds: ${timeoutSeconds}`);

    try {
        const response = await GetAddressInsightClient.invoke(
            businessName,
            address1,
            address2,
            city,
            state,
            zip,
            testType,
            licenseKey,
            isLive,
            timeoutSeconds
        );

        if (response.Error) {
            console.log("\n* Error *\n");
            console.log(`Error Type       : ${response.Error.Type}`);
            console.log(`Error Type Code  : ${response.Error.TypeCode}`);
            console.log(`Error Desc       : ${response.Error.Desc}`);
            console.log(`Error Desc Code  : ${response.Error.DescCode}`);
            return;
        }

        console.log("\n* Address Insight *\n");
        if (response) {
            console.log(`Status                 : ${response.Status}`);
            console.log(`Status Score           : ${response.StatusScore}`);
            console.log(`Address Status         : ${response.AddressStatus}`);
            console.log(`DPV                    : ${response.DPV}`);
            console.log(`DPV Description        : ${response.DPVDesc}`);
            console.log(`Address                : ${response.Address}`);
            console.log(`Address Extra          : ${response.AddressExtra}`);
            console.log(`City                   : ${response.City}`);
            console.log(`State                  : ${response.State}`);
            console.log(`Zip                    : ${response.Zip}`);
            console.log(`Barcode Digits         : ${response.BarcodeDigits}`);
            console.log(`Carrier Route          : ${response.CarrierRoute}`);
            console.log(`Congress Code          : ${response.CongressCode}`);
            console.log(`County Name            : ${response.CountyName}`);
            console.log(`Fragment House         : ${response.FragmentHouse}`);
            console.log(`Fragment PreDir        : ${response.FragmentPreDir}`);
            console.log(`Fragment Street        : ${response.FragmentStreet}`);
            console.log(`Fragment Suffix        : ${response.FragmentSuffix}`);
            console.log(`Fragment PostDir       : ${response.FragmentPostDir}`);
            console.log(`Fragment Unit          : ${response.FragmentUnit}`);
            console.log(`Fragment Unit Num      : ${response.FragmentUnitNumber}`);
            console.log(`Fragment PMB Prefix    : ${response.FragmentPMBPrefix}`);
            console.log(`Fragment PMB Number    : ${response.FragmentPMBNumber}`);
            console.log(`Corrections            : ${response.Corrections}`);
            console.log(`Corrections Desc       : ${response.CorrectionsDesc}`);
            console.log(`Address Notes          : ${response.AddressNotes}`);
            console.log(`Address Notes Codes    : ${response.AddressNotesCodes}`);
            console.log(`Geocode Status         : ${response.GeocodeStatus}`);
            console.log(`Latitude               : ${response.LocationLatitude}`);
            console.log(`Longitude              : ${response.LocationLongitude}`);
            console.log(`Census Tract           : ${response.CensusTract}`);
            console.log(`Census Block           : ${response.CensusBlock}`);
            console.log(`Place Name             : ${response.PlaceName}`);
            console.log(`Class FP               : ${response.ClassFP}`);
            console.log(`SLDUST                 : ${response.SLDUST}`);
            console.log(`SLDLST                 : ${response.SLDLST}`);
            console.log(`County FIPS            : ${response.CountyFIPS}`);
            console.log(`State FIPS             : ${response.StateFIPS}`);
            console.log(`Geocode Notes          : ${response.GeocodeNotes}`);
            console.log(`Geocode Notes Codes    : ${response.GeocodeNotesCodes}`);
            console.log(`Zip Status             : ${response.ZipStatus}`);
            console.log(`Zip Latitude           : ${response.ZipLatitude}`);
            console.log(`Zip Longitude          : ${response.ZipLongitude}`);
            console.log(`City Type              : ${response.CityType}`);
            console.log(`City Alias Name        : ${response.CityAliasName}`);
            console.log(`Area Code              : ${response.AreaCode}`);
            console.log(`Time Zone              : ${response.TimeZone}`);
            console.log(`Daylight Saving        : ${response.DaylightSaving}`);
            console.log(`MSA                    : ${response.MSA}`);
            console.log(`CBSA                   : ${response.CBSA}`);
            console.log(`CBSA Div               : ${response.CBSA_Div}`);
            console.log(`PMSA                   : ${response.PMSA}`);
            console.log(`DMA                    : ${response.DMA}`);
            console.log(`Zip Household Value    : ${response.ZipHouseholdValue}`);
            console.log(`Persons Per Household  : ${response.ZipPersonsPerHousehold}`);
            console.log(`Zip Household Income   : ${response.ZipHouseholdIncome}`);
            console.log(`County Household Income: ${response.CountyHouseholdIncome}`);
            console.log(`State Household Income : ${response.StateHouseholdIncome}`);
            console.log(`Zip Notes              : ${response.ZipNotes}`);
            console.log(`Zip Notes Codes        : ${response.ZipNotesCodes}`);
        } else {
            console.log("No address insight data found.");
        }

    } catch (e) {
        console.log("\n* Error *\n");
        console.log(`Error Message: ${e.message}`);
    }
}

export { GetAddressInsightClientGo };

File: address-insight-us-dot-net-examples/GetAddressInsightRestSdkExample.cs
using address_insight_us_dot_net.REST;

namespace address_insight_us_dot_net_examples
{
    public class GetAddressInsightRestSdkExample
    {
        public static void Go(string licenseKey, bool isLive)
        {
            Console.WriteLine("\r\n---------------------------------------------------");
            Console.WriteLine("Address Insight – US - GetAddressInsight - REST SDK");
            Console.WriteLine("---------------------------------------------------");

            GetAddressInsightClient.GetAddressInsightInput getAddressInsightInput = new(
                BusinessName: "Service Objects",
                Address1: "136 W Canon Perdido St, Suite D",
                Address2: "",
                City: "Santa Barbara",
                State: "CA",
                Zip: "93101",
                TestType: "census_loose",
                LicenseKey: licenseKey,
                IsLive: isLive,
                TimeoutSeconds: 15
            );

            Console.WriteLine("\r\n* Input *\r\n");
            Console.WriteLine($"Business Name: {getAddressInsightInput.BusinessName}");
            Console.WriteLine($"Address1     : {getAddressInsightInput.Address1}");
            Console.WriteLine($"Address2     : {getAddressInsightInput.Address2}");
            Console.WriteLine($"City         : {getAddressInsightInput.City}");
            Console.WriteLine($"State        : {getAddressInsightInput.State}");
            Console.WriteLine($"Zip          : {getAddressInsightInput.Zip}");
            Console.WriteLine($"Test Type    : {getAddressInsightInput.TestType}");
            Console.WriteLine($"License Key  : {getAddressInsightInput.LicenseKey}");
            Console.WriteLine($"Is Live      : {getAddressInsightInput.IsLive}");

            AINResponse response = GetAddressInsightClient.Invoke(getAddressInsightInput);
            if (response.Error is null)
            {
                Console.WriteLine("\r\n* Address Insight *\r\n");
                Console.WriteLine($"Status                 : {response.Status}");
                Console.WriteLine($"Status Score           : {response.StatusScore}");
                Console.WriteLine($"Address Status         : {response.AddressStatus}");
                Console.WriteLine($"DPV                    : {response.DPV}");
                Console.WriteLine($"DPV Description        : {response.DPVDesc}");
                Console.WriteLine($"Address                : {response.Address}");
                Console.WriteLine($"Address Extra          : {response.AddressExtra}");
                Console.WriteLine($"City                   : {response.City}");
                Console.WriteLine($"State                  : {response.State}");
                Console.WriteLine($"Zip                    : {response.Zip}");
                Console.WriteLine($"Barcode Digits         : {response.BarcodeDigits}");
                Console.WriteLine($"Carrier Route          : {response.CarrierRoute}");
                Console.WriteLine($"Congress Code          : {response.CongressCode}");
                Console.WriteLine($"County Name            : {response.CountyName}");
                Console.WriteLine($"Fragment House         : {response.FragmentHouse}");
                Console.WriteLine($"Fragment PreDir        : {response.FragmentPreDir}");
                Console.WriteLine($"Fragment Street        : {response.FragmentStreet}");
                Console.WriteLine($"Fragment Suffix        : {response.FragmentSuffix}");
                Console.WriteLine($"Fragment PostDir       : {response.FragmentPostDir}");
                Console.WriteLine($"Fragment Unit          : {response.FragmentUnit}");
                Console.WriteLine($"Fragment Unit Num      : {response.FragmentUnitNumber}");
                Console.WriteLine($"Fragment PMB Prefix    : {response.FragmentPMBPrefix}");
                Console.WriteLine($"Fragment PMB Number    : {response.FragmentPMBNumber}");
                Console.WriteLine($"Corrections            : {response.Corrections}");
                Console.WriteLine($"Corrections Desc       : {response.CorrectionsDesc}");
                Console.WriteLine($"Address Notes          : {response.AddressNotes}");
                Console.WriteLine($"Address Notes Codes    : {response.AddressNotesCodes}");
                Console.WriteLine($"Geocode Status         : {response.GeocodeStatus}");
                Console.WriteLine($"Latitude               : {response.LocationLatitude}");
                Console.WriteLine($"Longitude              : {response.LocationLongitude}");
                Console.WriteLine($"Census Tract           : {response.CensusTract}");
                Console.WriteLine($"Census Block           : {response.CensusBlock}");
                Console.WriteLine($"Place Name             : {response.PlaceName}");
                Console.WriteLine($"Class FP               : {response.ClassFP}");
                Console.WriteLine($"SLDUST                 : {response.SLDUST}");
                Console.WriteLine($"SLDLST                 : {response.SLDLST}");
                Console.WriteLine($"County FIPS            : {response.CountyFIPS}");
                Console.WriteLine($"State FIPS             : {response.StateFIPS}");
                Console.WriteLine($"Geocode Notes          : {response.GeocodeNotes}");
                Console.WriteLine($"Geocode Notes Codes    : {response.GeocodeNotesCodes}");
                Console.WriteLine($"Zip Status             : {response.ZipStatus}");
                Console.WriteLine($"Zip Latitude           : {response.ZipLatitude}");
                Console.WriteLine($"Zip Longitude          : {response.ZipLongitude}");
                Console.WriteLine($"City Type              : {response.CityType}");
                Console.WriteLine($"City Alias Name        : {response.CityAliasName}");
                Console.WriteLine($"Area Code              : {response.AreaCode}");
                Console.WriteLine($"Time Zone              : {response.TimeZone}");
                Console.WriteLine($"Daylight Saving        : {response.DaylightSaving}");
                Console.WriteLine($"MSA                    : {response.MSA}");
                Console.WriteLine($"CBSA                   : {response.CBSA}");
                Console.WriteLine($"CBSA Div               : {response.CBSA_Div}");
                Console.WriteLine($"PMSA                   : {response.PMSA}");
                Console.WriteLine($"DMA                    : {response.DMA}");
                Console.WriteLine($"Zip Household Value    : {response.ZipHouseholdValue}");
                Console.WriteLine($"Persons Per Household  : {response.ZipPersonsPerHousehold}");
                Console.WriteLine($"Zip Household Income   : {response.ZipHouseholdIncome}");
                Console.WriteLine($"County Household Income: {response.CountyHouseholdIncome}");
                Console.WriteLine($"State Household Income : {response.StateHouseholdIncome}");
                Console.WriteLine($"Zip Notes              : {response.ZipNotes}");
                Console.WriteLine($"Zip Notes Codes        : {response.ZipNotesCodes}");
            }
            else
            {
                Console.WriteLine("\r\n* Error *\r\n");
                Console.WriteLine($"Error Type        : {response.Error.Type}");
                Console.WriteLine($"Error Type Code   : {response.Error.TypeCode}");
                Console.WriteLine($"Error Desc        : {response.Error.Desc}");
                Console.WriteLine($"Error Desc Code   : {response.Error.DescCode}");
            }
        }
    }
}